services/collectibles/api.go

```diff
@@ -188,3 +217,50 @@ func (api *API) validateWalletsAndAmounts(walletAddresses []string, amount int)
 	}
 	return nil
 }
+
+func (api *API) EstimateRemoteBurn(ctx context.Context, chainID uint64, contractAddress string, tokenIds []*bigint.BigInt) (uint64, error) {
+	if len(tokenIds) == 0 {
+		return 0, errors.New("token list is empty")
+	}
+
+	ethClient, err := api.RPCClient.EthClient(chainID)
+	if err != nil {
+		log.Error(err.Error())
+		return 0, err
+	}
+
+	collectiblesABI, err := abi.JSON(strings.NewReader(collectibles.CollectiblesABI))
+	if err != nil {
+		return 0, err
+	}
+
+	var tempTokenIds []*big.Int
+	for _, v := range tokenIds {
+		tempTokenIds = append(tempTokenIds, v.Int)
+	}
+
+	data, err := collectiblesABI.Pack("remoteBurn", tempTokenIds)
+	if err != nil {
+		return 0, err
+	}
+
+	ownerAddr, err := api.ContractOwner(ctx, chainID, contractAddress)
+	if err != nil {
+		return 0, err
+	}
+	toAddr := common.HexToAddress(contractAddress)
+	fromAddr := common.HexToAddress(ownerAddr)
+
+	callMsg := ethereum.CallMsg{
+		From:  fromAddr,
+		To:    &toAddr,
+		Value: big.NewInt(0),
+		Data:  data,
+	}
+
+	estimate, err := ethClient.EstimateGas(ctx, callMsg)
+	if err != nil {
+		return 0, err
+	}
+	return estimate + uint64(float32(estimate)*0.1), nil
```

Comment:
Out of curiosity, why do we need this extra 10%?

Reply:
Typically, you want to add a bit of slack when doing gas estimation to increase chances your transaction goes through.
The estimate is likely a good average, but might diverge on a case by case bases (and depending on how high demand is)

Reply:
Exactly, this is a typical practice to add 10% to the estimated amount.